### PR TITLE
Dockerize web client (dev env)

### PR DESCRIPTION
**Pull request by materad**
Add docker support for web client development
Support hot modules reloading

### CODE REVIEW

File: WebClient/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM node:9.11-alpine
```
Comment (private0x0a) [inline WebClient/Dockerfile]: Why not lts?

File: WebClient/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+ENV NPM_CONFIG_LOGLEVEL warn
+
+WORKDIR /usr/basket-app
```
Comment (private0x0a) [inline WebClient/Dockerfile]: Use relative path

File: WebClient/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+COPY package.json package.json
+COPY yarn.lock  yarn.lock
+RUN yarn install
+
+COPY . .
```
Comment (private0x0a) [inline WebClient/Dockerfile]: Can you explain this?

File: docker-compose.yml [hunk 4]
```diff
@@ -1,6 +1,14 @@
 version: '3.3'
 
 services:
+  web:
+    build: ./WebClient/
+    command: yarn run start
```
Comment (private0x0a) [inline docker-compose.yml]: yarn start

File: docker-compose.yml [hunk 5]
```diff
@@ -1,6 +1,14 @@
...
+    volumes:
+      - './WebClient/src/:/usr/basket-app/src'
```
Comment (private0x0a) [inline docker-compose.yml]: Use relative path

File: WebClient/Dockerfile (hunk 1 above)
Comment (materad) [reply WebClient/Dockerfile]: There is no such image. https://hub.docker.com/r/_/node/

File: docker-compose.yml (hunk 4 above)
Comment (krzysztofla) [reply docker-compose.yml]: npm run start
yarn start

File: WebClient/Dockerfile (hunk 3 above)
Comment (materad) [reply WebClient/Dockerfile]: "layer creation can be leveraged to to cache portions of an application that rarely change, speeding up build times. This caching mechanism can be leveraged in a Node.js app - by caching its node_modules directory. By adding the app's package.json and running npm install before copying in the app's source files, npm install doesn't need to be run after every build."
A bit below there is a snippet from docker file:
COPY package.json .
RUN npm install --production
COPY . .

I read somewhere that adding yarn.lock to the container is a good idea to preserve last good configuration, but Im not a yarn expert, as you could have already observed with how I execute run script ;)
Anyway, more extensive explanation can be found here:
https://nodesource.com/blog/8-protips-to-start-killing-it-when-dockerizing-node-js/

File: docker-compose.yml (hunk 4 above)
Comment (materad) [reply docker-compose.yml]: Corrected

File: docker-compose.yml (hunk 5 above)
Comment (materad) [reply docker-compose.yml]: volumes host os path accepts only absolute path

**Pull request merged**